src/main/kotlin/de/uniwuerzburg/omod/core/Omod.kt

```diff
@@ -378,18 +391,24 @@ class Omod(
             val homeZoneID = sampleCumDist(homeCumDist, rng)
             val homeZone = zones[homeZoneID]
 
-            // Get home location
-            val home = if (homeZone is Cell) { // Home is building
-                val buildingsHomeDist = createCumDist(
-                    getHomeWeightsRestricted(homeZone.buildings, inside).toDoubleArray()
-                )
-                homeZone.buildings[sampleCumDist(buildingsHomeDist, rng)]
-            } else { // IS dummy location
-                homeZone
+
+            if (!tazFileAvailable) {
+                // Get home location as Building
+                val home = if (homeZone is Cell) { // Home is building
+                    val buildingsHomeDist = createCumDist(
+                        getHomeWeightsRestricted(homeZone.buildings, inside).toDoubleArray()
+                    )
+                    homeZone.buildings[sampleCumDist(buildingsHomeDist, rng)]
+                } else { // IS dummy location
+                    homeZone
+                }
+                agents.add(agentFactory.createAgent(home, homeZone))
+
+            } else {
+                // Get home location as TAZ
+                agents.add(agentFactory.createAgent(homeZone, homeZone))
```

Comment:
This is a different feature. Should be called "Approximate mode" or something. TAZ file option should also work with detailed simulations.